src/sequences/OEISSequenceTemplate.ts

```diff
@@ -50,14 +54,17 @@ export default class OEISSequenceTemplate extends SequenceCached {
         if (this.settings['oeisId'] !== undefined) {
             this.isValid = true;
             if (this.settings['numElements']) {
-                this.entries = Number(this.settings['numElements']);
+                this.last = Number(this.settings['numElements']) - 1;
+            } else {
+                this.last = 999;
             }
+            this.blocksize = this.last + 1; // get everything on the initial fill
             this.name = (this.settings['name'] as string);
             return new ValidationStatus(true);
         }
 
         this.isValid = false;
-        return new ValidationStatus(false, ["oeisID paramter is missing"]);
+        return new ValidationStatus(false, ["oeisID parameter is missing"]);
```

Comment:
Hmmm; as counterpoint, I find generally that exiting from a function as soon as possible makes code more streamlined and much more comprehensible. So I frequently write code with many returns; to me, they're sort of like the "break"s in switch() statements, and keep the blocks that handle different things nice and clearly separate. i.e., instead of 
``` myfun(x) {
 retval = default;
 if (x is like this) {
   retval = 7;
} else if (x is like that) {
   retval = 9;
}
return retval;
}
```
I would be inclined to write
```
myfun(x) {
  if (x is like this) return 7;
  if (x is like that) return 9;
  return default;
}
```
How should we reconcile this coding difference? Or should we simply live with somewhat different forms of capturing the desired behavior, as long as the code remains reasonably clear?

Reply:
I did read some stackexchange questions about single/multiple return points from functions. I feel like the most common pragmatic viewpoint was that some "guard-style" early returns and one main code path is often useful. With that in mind, I inverted the test on the presence of the oeisId, so that the function now has two simple early exits (if the super.validate() fails or if there is no oeisId, and then otherwise flows to the bottom for the main "success" return. Hopefully that's an agreeable compromise. (In general in validate(), it can be useful to accumulate all of the problems one has found into a list, and return with a false ValidationStatus if that list is non-empty, otherwise post-processing and returning a true Validation status in the end. I still don't see how it's better/more readable to avoid two returns at the cost of putting all of the post-processing inside a conditional. This is the comparison between:
```
  ...
  checkForErrors();
  if (thereWereErrors) return ValidationStatus(false,...);
  doYourStuff();
  return ValidationStatus(true, ....);
}
```
and
```
  ...
  validationStatus = checkForErrors();
  if (validationStatus.valid = true) {
     doYourStuff();
  }
  return validationStatus;
}
```
I guess they don't look so different above, but the point is that typically "doYourStuff()" is actually several lines of code, so the difference is between a simple exit as soon as you realize there's nothing more to do, versus having the "primary" code path that will be taken in "normal" conditions wrapped in a conditional. To me the latter is harder to read, and I am not clear on the benefit that having only one return statement provides in exchange.